src/DbAuth.cc

```diff
@@ -62,97 +60,79 @@ namespace pexPolicy = lsst::pex::policy;
 
 static pexPolicy::Policy::Ptr authPolicy(static_cast<pexPolicy::Policy*>(0));
 
-static std::pair<std::string, std::string>
-search(std::string const& host, std::string const& port) {
+static std::pair<std::string, std::string> search(std::string const& host, std::string const& port) {
     if (authPolicy == 0) {
         passwd pwd;
-        passwd *pw;
+        passwd* pw;
         long maxbuf = sysconf(_SC_GETPW_R_SIZE_MAX);
         std::unique_ptr<char[]> buffer(new char[maxbuf]);
         int ret = getpwuid_r(geteuid(), &pwd, buffer.get(), maxbuf, &pw);
         if (ret != 0 || pw->pw_dir == 0) {
-            throw LSST_EXCEPT(pexExcept::RuntimeError,
-                    "Could not get home directory");
+            throw LSST_EXCEPT(pexExcept::RuntimeError, "Could not get home directory");
         }
         std::string dir = std::string(pw->pw_dir) + "/.lsst";
         std::string filename = dir + "/db-auth.paf";
         struct stat st;
         ret = stat(dir.c_str(), &st);
         if (ret != 0 || (st.st_mode & (S_IRWXG | S_IRWXO)) != 0) {
-            throw LSST_EXCEPT(pexExcept::RuntimeError,
-                    dir + " directory is missing or accessible by others");
+            throw LSST_EXCEPT(pexExcept::RuntimeError, dir + " directory is missing or accessible by others");
         }
         ret = stat(filename.c_str(), &st);
         if (ret != 0 || (st.st_mode & (S_IRWXG | S_IRWXO)) != 0) {
-            throw LSST_EXCEPT(pexExcept::RuntimeError,
-                    filename + " is missing or accessible by others");
-        }
-        {
-            dafBase::PersistentCitizenScope scopeGuard;
-            authPolicy = pexPolicy::Policy::Ptr(new pexPolicy::Policy(filename));
+            throw LSST_EXCEPT(pexExcept::RuntimeError, filename + " is missing or accessible by others");
         }
+        { authPolicy = pexPolicy::Policy::Ptr(new pexPolicy::Policy(filename)); }
```

Comment:
Why is this happening in its own execution block?

Reply:
Ha! Look at the diff above: this was just collapsed by clang-format from lines 89-94 in the original.

So, I have no idea why it's a separate block.

Reply:
I think I vaguely recall either Jim or Pim working on something to to with the scopeGaurd. I think it was related to a ticket Pim had worked on to make daf persistence thread safe, which is what the scope had to be in place. Now that that is removed I think the scope block can be removed too.

Reply:
You may be right about the reasoning, but according to git history, the `scopeGuard` was added 10 years ago by `smm`.

https://github.com/lsst/daf_persistence/commit/58860a4a4f0a7d5db62405f16e1edc305eb56e23

I'm tempted to just leave it as is, given the imminent removal of `daf_persistence`. ;-)